codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -75,31 +77,29 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
 
     override fun longShape(shape: LongShape): Symbol = numberShape(shape, "Long", "0L")
 
-    override fun floatShape(shape: FloatShape): Symbol = numberShape(shape, "Float", "0.0f")
+    override fun floatShape(shape: FloatShape): Symbol = numberShape(shape, "Float", "0f")
 
     override fun doubleShape(shape: DoubleShape): Symbol = numberShape(shape, "Double", "0.0")
 
     private fun numberShape(shape: Shape, typeName: String, defaultValue: String = "0"): Symbol =
-        createSymbolBuilder(shape, typeName, namespace = "kotlin")
-            .defaultValue(defaultValue)
-            .build()
+        createSymbolBuilder(shape, typeName, namespace = "kotlin").defaultValue(defaultValue).build()
```

Comment:
The motivation behind INFERRED / MODELED is because we're using `.defaultValue()` in two ways now. The previous way is inferring that numbers and booleans have a default value (0, 0f, 0.0, false). The new way we use `.defaultValue()` is when it's modeled on a shape.

If a shape has an `INFERRED` default value but it's nullable, we set it to null. If a shape has a `MODELED` default value but it's nullable, we will use the default value.